com.unity.perception/Runtime/GroundTruth/Labelers/BoundingBoxLabeler.cs

```diff
@@ -108,8 +116,73 @@ void OnRenderedObjectInfosCalculated(int frameCount, NativeArray<RenderedObjectI
                     };
                 }
 
+                if (!CaptureOptions.useAsyncReadbackIfSupported && frameCount != Time.frameCount) 
+                    Debug.LogWarning("Not on current frame: " + frameCount + "(" + Time.frameCount + ")");
+
+                if (IsVisualizationEnabled()) 
+                {
+                    Visualize();
+                    ReportFrameCount(frameCount);
+                }
+                
                 asyncAnnotation.ReportValues(m_BoundingBoxValues);
             }
         }
+
+        /// <inheritdoc/>
+        protected override void SetupVisualizationPanel(GameObject panel)
+        {
+            var toggle  = GameObject.Instantiate(Resources.Load<GameObject>("GenericToggle"));
+            toggle.transform.SetParent(panel.transform);
+            toggle.GetComponentInChildren<Text>().text = "Bounding Boxes";
+            toggle.GetComponent<Toggle>().onValueChanged.AddListener(enabled => {
+                EnableVisualization(enabled);
+            });
+
+            visualizationHolder = GameObject.Instantiate(Resources.Load<GameObject>("BoundsHolder"));
+            visualizationHolder.transform.SetParent(panel.transform.parent, false);
```

Comment:
`panel.transform.parent` is used in a couple places as the container for visualizations. Should this be exposed explicitly?

Reply:
These are just Unity UI classes that I haven't put custom scripts on. So, I'm kind of doing it the Unity way, but maybe there is a better way that I don't know about? I always chalk this up to some of the ugly parts of dynamic UI scripting.

Reply:
When I look at this from an API contract perspective, it looks like the user needs to know that the parent of the panel is a canvas/other container that they can add UI to. That's a fairly weak contract. It would be nice to expose this more explicitly to make that contract more clear. It would also let us change around the hierarchy of the UI if we need to.

Reply:
This has all been reworked